# After doing the modeling and exploring some of the images, it became clear to me that some of the images have more than type of clothing shown. Also, some appear to be mislabeled. 

In [78]:
import pandas as pd
import numpy as np
import re
import seaborn as sns
import matplotlib.pyplot as plt

In [79]:
df = pd.read_json('/Users/butler/Documents/Image-Classification-Indian-Clothing/data/raw/train_data.json', lines=True)

In [80]:
df.head()

image_url           image_path  \
0  https://m.media-amazon.com/images/I/81XKaSKvly...  images/train/0.jpeg   
1  https://m.media-amazon.com/images/I/91dDMDg+QT...  images/train/1.jpeg   
2  https://m.media-amazon.com/images/I/71TJ9p46PV...  images/train/2.jpeg   
3  https://m.media-amazon.com/images/I/81Y+je7CEg...  images/train/3.jpeg   
4  https://m.media-amazon.com/images/I/61B8o9Ulqp...  images/train/4.jpeg   

         brand                                      product_title class_label  \
0    Womanista  Women's Georgette Saree with Blouse Piece (TKI...       saree   
1      Yashika         Women Linen Printed Sarees Free Size (BEE)       saree   
2    Womanista  Organza Floral Print Saree with Blouse Piece(W...       saree   
3  PERFECTBLUE  Women's Digital Cotton Linen Blend Saree with ...       saree   
4   VAIVIDHYAM  Designer Sarees Women's Banarasi Cotton Silk S...       saree   

  color  
0   NaN  
1   NaN  
2   NaN  
3   NaN  
4   NaN

In [81]:
df.tail()

image_url  \
91161  https://rukminim1.flixcart.com/image/309/371/k...   
91162  https://rukminim1.flixcart.com/image/309/371/k...   
91163  https://rukminim1.flixcart.com/image/309/371/k...   
91164  https://rukminim1.flixcart.com/image/309/371/k...   
91165  https://rukminim1.flixcart.com/image/309/371/k...   

                    image_path           brand  \
91161  images/train/91161.jpeg         AURELIA   
91162  images/train/91162.jpeg  Prime clothing   
91163  images/train/91163.jpeg           Vrisa   
91164  images/train/91164.jpeg   HELLEN VELLEN   
91165  images/train/91165.jpeg     NEW STYLISH   

                                        product_title  class_label       color  
91161        Women Printed Pure Cotton Straight Kurta  women_kurta        Blue  
91162     Women Checkered Cotton Blend Straight Kurta  women_kurta       White  
91163      Women Embroidered Georgette Straight Kurta  women_kurta       White  
91164                Women Printed Rayon A-line Kurta  women_kurta   Dark Blue  
91165  Women Printed Rayon Crepe Blend Straight Kurta  women_kurta  Dark Green

In [82]:
df.shape

(91166, 6)

In [83]:
df.class_label.unique()

array(['saree', 'blouse', 'dhoti_pants', 'dupattas', 'gowns', 'kurta_men',
       'leggings_and_salwars', 'lehenga', 'mojaris_men', 'mojaris_women',
       'nehru_jackets', 'palazzos', 'petticoats', 'sherwanis',
       'women_kurta'], dtype=object)

In [84]:
df_revised = df.copy()

In [85]:
df_revised.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91166 entries, 0 to 91165
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   image_url      91166 non-null  object
 1   image_path     91166 non-null  object
 2   brand          91166 non-null  object
 3   product_title  91166 non-null  object
 4   class_label    91166 non-null  object
 5   color          10002 non-null  object
dtypes: object(6)
memory usage: 4.2+ MB


In [86]:
# Resetting index in case the original indices are needed
df_revised = df_revised.reset_index(names='number')

In [87]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color  
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN  
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN  
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN  
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN  
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN

In [88]:
df_revised.tail()

number                                          image_url  \
91161   91161  https://rukminim1.flixcart.com/image/309/371/k...   
91162   91162  https://rukminim1.flixcart.com/image/309/371/k...   
91163   91163  https://rukminim1.flixcart.com/image/309/371/k...   
91164   91164  https://rukminim1.flixcart.com/image/309/371/k...   
91165   91165  https://rukminim1.flixcart.com/image/309/371/k...   

                    image_path           brand  \
91161  images/train/91161.jpeg         AURELIA   
91162  images/train/91162.jpeg  Prime clothing   
91163  images/train/91163.jpeg           Vrisa   
91164  images/train/91164.jpeg   HELLEN VELLEN   
91165  images/train/91165.jpeg     NEW STYLISH   

                                        product_title  class_label       color  
91161        Women Printed Pure Cotton Straight Kurta  women_kurta        Blue  
91162     Women Checkered Cotton Blend Straight Kurta  women_kurta       White  
91163      Women Embroidered Georgette Straight Kurta  women_kurta       White  
91164                Women Printed Rayon A-line Kurta  women_kurta   Dark Blue  
91165  Women Printed Rayon Crepe Blend Straight Kurta  women_kurta  Dark Green

In [89]:
# Using column names with smallest part of clothing name
columns = ['kurta', 'women', 'men', 'dhoti', 'saree', 'sherwani', 'palazzo', 
           'mojari', 'nehru', 'legging', 'salwar', 'petticoat', 'blouse', 
           'dupatta', 'gown','lehenga']


In [90]:
# Adding these columns to the df with the values being true if the product title contains the column name
for column in columns:
    df_revised[str(column)] = df_revised.product_title.str.contains(str(column), flags=re.IGNORECASE)

In [91]:
# Checking for rows that have girl in the product title
df_revised['girl'] = df_revised.product_title.str.contains('girl', flags=re.IGNORECASE)

In [92]:
# Putting the rows that have girl in the product title into the women category
c1 = (df_revised.women == False)
c2 = (df_revised.girl == True)

df_revised.loc[c2 & c2, 'women'] = True

In [93]:
# Checking for rows that have woman in the product title
df_revised['woman'] = df_revised.product_title.str.contains('woman', flags=re.IGNORECASE)

In [94]:
# Putting the rows that have woman in the product title into the women category
c1 = (df_revised.women == False)
c2 = (df_revised.woman == True)

df_revised.loc[c2 & c2, 'women'] = True

In [95]:
# Checking for rows that have boy in the product title
df_revised['man'] = df_revised.product_title.str.contains('man', flags=re.IGNORECASE)

In [96]:
# Checking for rows that have man in the product title
df_revised['boy'] = df_revised.product_title.str.contains('boy', flags=re.IGNORECASE)

In [97]:
# Putting the rows that have man in the product title into the men category
c1 = (df_revised.men == False)
c2 = (df_revised.man == True)

df_revised.loc[c2 & c2, 'men'] = True

In [98]:
# Putting the rows that have boy in the product title into the men category
c1 = (df_revised.men == False)
c2 = (df_revised.boy == True)

df_revised.loc[c2 & c2, 'men'] = True

In [99]:
# No longer needed
df_revised.drop(['girl', 'woman', 'man', 'boy'], axis=1, inplace=True)
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color  kurta  \
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN  False   
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN  False   
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN  False   
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN  False   
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN  False   

   women    men  ...  palazzo  mojari  nehru  legging  salwar  petticoat  \
0   True   True  ...    False   False  False    False   False      False   
1   True   True  ...    False   False  False    False   False      False   
2  False  False  ...    False   False  False    False   False      False   
3   True   True  ...    False   False  False    False   False      False   
4   True   True  ...    False   False  False    False   False      False   

   blouse  dupatta   gown  lehenga  
0    True    False  False    False  
1   False    False  False    False  
2    True    False  False    False  
3    True    False  False    False  
4    True    False  False    False  

[5 rows x 23 columns]

In [100]:
df_revised.columns

Index(['number', 'image_url', 'image_path', 'brand', 'product_title',
       'class_label', 'color', 'kurta', 'women', 'men', 'dhoti', 'saree',
       'sherwani', 'palazzo', 'mojari', 'nehru', 'legging', 'salwar',
       'petticoat', 'blouse', 'dupatta', 'gown', 'lehenga'],
      dtype='object')

In [101]:
# If the columns entitled 'men' and 'women' are both true, changing the 'men' column to false
c1 = (df_revised['men'] == True) 
c2 = (df_revised['women'] == True)

df_revised.loc[c1 & c2, 'men'] = False

In [102]:
# Making a column for women's kurtas
df_revised['women_kurta'] = [False]* 91166

In [103]:
# Making a column for men's kurtas
df_revised['kurta_men'] = [False]* 91166

In [104]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color  kurta  \
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN  False   
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN  False   
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN  False   
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN  False   
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN  False   

   women    men  ...  nehru  legging  salwar  petticoat  blouse  dupatta  \
0   True  False  ...  False    False   False      False    True    False   
1   True  False  ...  False    False   False      False   False    False   
2  False  False  ...  False    False   False      False    True    False   
3   True  False  ...  False    False   False      False    True    False   
4   True  False  ...  False    False   False      False    True    False   

    gown  lehenga  women_kurta  kurta_men  
0  False    False        False      False  
1  False    False        False      False  
2  False    False        False      False  
3  False    False        False      False  
4  False    False        False      False  

[5 rows x 25 columns]

In [105]:
# Making women_kurta true and kurta false if kurta is true and women is true
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised['women'] == True)

df_revised.loc[c1 & c2, 'women_kurta'] = True
df_revised.loc[c1 & c2, 'kurta'] = False

In [106]:
# Making kurta_men true and kurta false if kurta is true and men is true
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised['men'] == True)

df_revised.loc[c1 & c2, 'kurta_men'] = True
df_revised.loc[c1 & c2, 'kurta'] = False

In [107]:
# Making kurta false when if kurta is true and kurta_men is the class label
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised.class_label == 'kurta_men')

df_revised.loc[c1 & c2, 'kurta'] = False

In [108]:
# Making kurta false when if kurta is true and women_kurta is the class label
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised.class_label == 'women_kurta')

df_revised.loc[c1 & c2, 'kurta'] = False

In [109]:
(df_revised.kurta == True).value_counts()

kurta
False    91028
True       138
Name: count, dtype: int64

### This leaves 138 rows where kurta is true, men and women are false, and the class label does not contain kurta

In [110]:
# Since only women wear dupattas, if the class_label is dupatta and kurta is true, is must
# be a women's kurta
c1 = (df_revised.class_label == 'dupattas')
c2 = (df_revised.kurta == True)

df_revised.loc[c1 & c2, 'women_kurta'] = True

In [111]:
# Making columns for mojaris_men, mojaris_women, and leggings_and_salwars
df_revised['mojaris_men'] = [False]*91166
df_revised['mojaris_women'] = [False]*91166
df_revised['leggings_and_salwars'] = [False]*91166

In [112]:
# mojaris_women column is true if mojaris and women are true
c1 = (df_revised['mojari'] == True) 
c2 = (df_revised['women'] == True)

df_revised.loc[c1 & c2, 'mojaris_women'] = True

In [113]:
# mojaris_men column is true if mojaris and men are true
c1 = (df_revised['mojari'] == True) 
c2 = (df_revised['men'] == True)

df_revised.loc[c1 & c2, 'mojaris_men'] = True

In [114]:
# Putting the columns legging and salwar together
c1 = (df_revised['legging'] == True) 
c2 = (df_revised['salwar'] == True)

df_revised.loc[c1 | c2, 'leggings_and_salwars'] = True

In [115]:
# Dropping the columns that are no longer needed
df_revised.drop(['men', 'women', 'legging','salwar','mojari'], axis=1, inplace=True)

In [116]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color  kurta  \
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN  False   
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN  False   
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN  False   
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN  False   
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN  False   

   dhoti  saree  ...  petticoat  blouse  dupatta   gown  lehenga  women_kurta  \
0  False   True  ...      False    True    False  False    False        False   
1  False   True  ...      False   False    False  False    False        False   
2  False   True  ...      False    True    False  False    False        False   
3  False   True  ...      False    True    False  False    False        False   
4  False   True  ...      False    True    False  False    False        False   

   kurta_men  mojaris_men  mojaris_women  leggings_and_salwars  
0      False        False          False                 False  
1      False        False          False                 False  
2      False        False          False                 False  
3      False        False          False                 False  
4      False        False          False                 False  

[5 rows x 23 columns]

In [117]:
# Removing the blouse label from all sarees in class_label because 
# models wearing sarees cannot be shown without a saree blouse

c1 = (df_revised['class_label'] == 'saree') 
c2 = (df_revised['blouse'] == True)

df_revised.loc[c1 & c2, 'blouse'] = False

In [118]:
# Removing the saree label from all blouses in class_label because 
# blouses are blouses for sarees or lehengas. Therefore, it is likely that saree 
# will be in the product description

c1 = (df_revised['class_label'] == 'blouse') 
c2 = (df_revised['saree'] == True)

df_revised.loc[c1 & c2, 'saree'] = False

In [119]:
# Removing the saree label from all blouses in class_label because 
# blouses are blouses for sarees or lehengas.Therefore, it is likely that lehenga 
# will be in the product description

c1 = (df_revised['class_label'] == 'blouse') 
c2 = (df_revised['lehenga'] == True)

df_revised.loc[c1 & c2, 'lehenga'] = False

In [120]:
# Removing the saree label from all petticoats in class_label because petticoats are worn under sarees. 
c1 = (df_revised['class_label'] == 'petticoat') 
c2 = (df_revised['saree'] == True)

df_revised.loc[c1 & c2, 'saree'] = False

In [121]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color  kurta  \
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN  False   
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN  False   
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN  False   
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN  False   
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN  False   

   dhoti  saree  ...  petticoat  blouse  dupatta   gown  lehenga  women_kurta  \
0  False   True  ...      False   False    False  False    False        False   
1  False   True  ...      False   False    False  False    False        False   
2  False   True  ...      False   False    False  False    False        False   
3  False   True  ...      False   False    False  False    False        False   
4  False   True  ...      False   False    False  False    False        False   

   kurta_men  mojaris_men  mojaris_women  leggings_and_salwars  
0      False        False          False                 False  
1      False        False          False                 False  
2      False        False          False                 False  
3      False        False          False                 False  
4      False        False          False                 False  

[5 rows x 23 columns]

In [122]:
df_revised.loc[0]

number                                                                  0
image_url               https://m.media-amazon.com/images/I/81XKaSKvly...
image_path                                            images/train/0.jpeg
brand                                                           Womanista
product_title           Women's Georgette Saree with Blouse Piece (TKI...
class_label                                                         saree
color                                                                 NaN
kurta                                                               False
dhoti                                                               False
saree                                                                True
sherwani                                                            False
palazzo                                                             False
nehru                                                               False
petticoat                             

In [123]:
# Renaming columns to conform to class labels
df_revised = df_revised.rename({'dhoti':'dhoti_pants', 'sherwani':'sherwanis', 'palazzo':'palazzos', 
             'nehru':'nehru_jackets', 'petticoat':'petticoats', 'dupatta':'dupattas', 
             'gown':'gowns'},axis=1)
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color  kurta  \
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN  False   
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN  False   
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN  False   
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN  False   
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN  False   

   dhoti_pants  saree  ...  petticoats  blouse  dupattas  gowns  lehenga  \
0        False   True  ...       False   False     False  False    False   
1        False   True  ...       False   False     False  False    False   
2        False   True  ...       False   False     False  False    False   
3        False   True  ...       False   False     False  False    False   
4        False   True  ...       False   False     False  False    False   

   women_kurta  kurta_men  mojaris_men  mojaris_women  leggings_and_salwars  
0        False      False        False          False                 False  
1        False      False        False          False                 False  
2        False      False        False          False                 False  
3        False      False        False          False                 False  
4        False      False        False          False                 False  

[5 rows x 23 columns]

In [124]:
# Changing each new column to false if the class_label is the same as the column name
columns = ['dhoti_pants', 'saree', 'sherwanis', 'palazzos', 'nehru_jackets', 'petticoats',
           'blouse','dupattas', 'gowns', 'lehenga', 'women_kurta', 'kurta_men', 
           'mojaris_men', 'mojaris_women', 'leggings_and_salwars']

for column in columns:
    c1 = (df_revised[str(column)] == True)
    c2 = (df_revised['class_label'] == str(column))
    df_revised.loc[c1 & c2, str(column)] = False

df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color  kurta  \
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN  False   
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN  False   
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN  False   
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN  False   
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN  False   

   dhoti_pants  saree  ...  petticoats  blouse  dupattas  gowns  lehenga  \
0        False  False  ...       False   False     False  False    False   
1        False  False  ...       False   False     False  False    False   
2        False  False  ...       False   False     False  False    False   
3        False  False  ...       False   False     False  False    False   
4        False  False  ...       False   False     False  False    False   

   women_kurta  kurta_men  mojaris_men  mojaris_women  leggings_and_salwars  
0        False      False        False          False                 False  
1        False      False        False          False                 False  
2        False      False        False          False                 False  
3        False      False        False          False                 False  
4        False      False        False          False                 False  

[5 rows x 23 columns]

In [125]:
columns = ['dhoti_pants', 'saree', 'kurta', 'sherwanis', 'palazzos', 'nehru_jackets', 'petticoats',
           'blouse','dupattas', 'gowns', 'lehenga', 'women_kurta', 'kurta_men', 
           'mojaris_men', 'mojaris_women', 'leggings_and_salwars']
# If a column is true, putting the column name as the value; otherwise, 0
def func(x):
    if x == True:
        return str(column)
    else:
        return 0

for column in columns:
    df_revised[str(column)] = df_revised[str(column)].apply(lambda x: func(x))

In [126]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/81XKaSKvly...   
1       1  https://m.media-amazon.com/images/I/91dDMDg+QT...   
2       2  https://m.media-amazon.com/images/I/71TJ9p46PV...   
3       3  https://m.media-amazon.com/images/I/81Y+je7CEg...   
4       4  https://m.media-amazon.com/images/I/61B8o9Ulqp...   

            image_path        brand  \
0  images/train/0.jpeg    Womanista   
1  images/train/1.jpeg      Yashika   
2  images/train/2.jpeg    Womanista   
3  images/train/3.jpeg  PERFECTBLUE   
4  images/train/4.jpeg   VAIVIDHYAM   

                                       product_title class_label color kurta  \
0  Women's Georgette Saree with Blouse Piece (TKI...       saree   NaN     0   
1         Women Linen Printed Sarees Free Size (BEE)       saree   NaN     0   
2  Organza Floral Print Saree with Blouse Piece(W...       saree   NaN     0   
3  Women's Digital Cotton Linen Blend Saree with ...       saree   NaN     0   
4  Designer Sarees Women's Banarasi Cotton Silk S...       saree   NaN     0   

  dhoti_pants saree  ... petticoats blouse dupattas gowns lehenga women_kurta  \
0           0     0  ...          0      0        0     0       0           0   
1           0     0  ...          0      0        0     0       0           0   
2           0     0  ...          0      0        0     0       0           0   
3           0     0  ...          0      0        0     0       0           0   
4           0     0  ...          0      0        0     0       0           0   

  kurta_men mojaris_men mojaris_women leggings_and_salwars  
0         0           0             0                    0  
1         0           0             0                    0  
2         0           0             0                    0  
3         0           0             0                    0  
4         0           0             0                    0  

[5 rows x 23 columns]

In [127]:
df_revised.tail()

number                                          image_url  \
91161   91161  https://rukminim1.flixcart.com/image/309/371/k...   
91162   91162  https://rukminim1.flixcart.com/image/309/371/k...   
91163   91163  https://rukminim1.flixcart.com/image/309/371/k...   
91164   91164  https://rukminim1.flixcart.com/image/309/371/k...   
91165   91165  https://rukminim1.flixcart.com/image/309/371/k...   

                    image_path           brand  \
91161  images/train/91161.jpeg         AURELIA   
91162  images/train/91162.jpeg  Prime clothing   
91163  images/train/91163.jpeg           Vrisa   
91164  images/train/91164.jpeg   HELLEN VELLEN   
91165  images/train/91165.jpeg     NEW STYLISH   

                                        product_title  class_label  \
91161        Women Printed Pure Cotton Straight Kurta  women_kurta   
91162     Women Checkered Cotton Blend Straight Kurta  women_kurta   
91163      Women Embroidered Georgette Straight Kurta  women_kurta   
91164                Women Printed Rayon A-line Kurta  women_kurta   
91165  Women Printed Rayon Crepe Blend Straight Kurta  women_kurta   

            color kurta dhoti_pants saree  ... petticoats blouse dupattas  \
91161        Blue     0           0     0  ...          0      0        0   
91162       White     0           0     0  ...          0      0        0   
91163       White     0           0     0  ...          0      0        0   
91164   Dark Blue     0           0     0  ...          0      0        0   
91165  Dark Green     0           0     0  ...          0      0        0   

      gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
91161     0       0           0         0           0             0   
91162     0       0           0         0           0             0   
91163     0       0           0         0           0             0   
91164     0       0           0         0           0             0   
91165     0       0           0         0           0             0   

      leggings_and_salwars  
91161                    0  
91162                    0  
91163                    0  
91164                    0  
91165                    0  

[5 rows x 23 columns]

In [128]:
# Number of gowns column that have gowns as value
df_revised.gowns.value_counts()

gowns
0        90982
gowns      184
Name: count, dtype: int64

In [129]:
# Gowns are sometimes called dresses
df_revised['dress'] = df_revised.product_title.str.contains('dress', flags=re.IGNORECASE)

In [130]:
# If dress is true and gowns is not (0), put the value gowns in from gowns column
c1 = (df_revised.gowns == 0)
c2 = (df_revised.dress == True)

df_revised.loc[c1 & c2, 'gowns'] = 'gowns'

df_revised.gowns.value_counts()

gowns
0        89187
gowns     1979
Name: count, dtype: int64

In [131]:
# Removing columns not needed
df_revised = df_revised.drop(['number', 'image_url', 'brand', 'color', 'dress'], axis=1)

### Making a dataframe for each label that is different than the class label and exploring the product titles and class labels of some of the rows

In [132]:
df_saree = df_revised[df_revised.saree == 'saree']
df_saree.shape

(1723, 19)

In [133]:
df_saree.head()

image_path  \
20610  images/train/20610.jpeg   
21180  images/train/21180.jpeg   
21900  images/train/21900.jpeg   
22385  images/train/22385.jpeg   
24793  images/train/24793.jpeg   

                                           product_title  class_label kurta  \
20610  Jaipuri Fashion Women's Traditional Dhoti Pant...  dhoti_pants     0   
21180  Women's Designer Printed Rayon Dark Green Sare...  dhoti_pants     0   
21900  Blue Chanderi Pant Saree without Blouse (Fully...  dhoti_pants     0   
22385  Women's Woven Handloom Cotton Silk Saree with ...  dhoti_pants     0   
24793                       Women's Cotton Dupatta Saree     dupattas     0   

      dhoti_pants  saree sherwanis palazzos nehru_jackets petticoats  blouse  \
20610           0  saree         0        0             0          0  blouse   
21180           0  saree         0        0             0          0       0   
21900           0  saree         0        0             0          0  blouse   
22385           0  saree         0        0             0          0  blouse   
24793           0  saree         0        0             0          0       0   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
20610        0     0       0           0         0           0             0   
21180        0     0       0           0         0           0             0   
21900        0     0       0           0         0           0             0   
22385        0     0       0           0         0           0             0   
24793        0     0       0           0         0           0             0   

      leggings_and_salwars  
20610                    0  
21180                    0  
21900                    0  
22385                    0  
24793                    0

In [134]:
df_saree.iloc[0,1], df_saree.iloc[0,2]

("Jaipuri Fashion Women's Traditional Dhoti Pant Ready to Wear Saree with Blouse (Ombre Green)",
 'dhoti_pants')

### Mislabeled:

In [135]:
df_saree.iloc[3,1], df_saree.iloc[3,2]

("Women's Woven Handloom Cotton Silk Saree with Blouse Piece (Multicolor)",
 'dhoti_pants')

In [136]:
df_blouse = df_revised[df_revised.blouse == 'blouse']
df_blouse.shape

(2337, 19)

In [137]:
df_blouse.head()

image_path  \
20610  images/train/20610.jpeg   
21900  images/train/21900.jpeg   
22006  images/train/22006.jpeg   
22034  images/train/22034.jpeg   
22147  images/train/22147.jpeg   

                                           product_title  class_label kurta  \
20610  Jaipuri Fashion Women's Traditional Dhoti Pant...  dhoti_pants     0   
21900  Blue Chanderi Pant Saree without Blouse (Fully...  dhoti_pants     0   
22006       Womens Raglan Sleeve Body Blouse Nursing Top  dhoti_pants     0   
22034               Womens Raglan Sleeve Body Blouse Top  dhoti_pants     0   
22147                Girls Raglan Sleeve Body Blouse Top  dhoti_pants     0   

      dhoti_pants  saree sherwanis palazzos nehru_jackets petticoats  blouse  \
20610           0  saree         0        0             0          0  blouse   
21900           0  saree         0        0             0          0  blouse   
22006           0      0         0        0             0          0  blouse   
22034           0      0         0        0             0          0  blouse   
22147           0      0         0        0             0          0  blouse   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
20610        0     0       0           0         0           0             0   
21900        0     0       0           0         0           0             0   
22006        0     0       0           0         0           0             0   
22034        0     0       0           0         0           0             0   
22147        0     0       0           0         0           0             0   

      leggings_and_salwars  
20610                    0  
21900                    0  
22006                    0  
22034                    0  
22147                    0

In [138]:
df_blouse.tail()

image_path  \
81355  images/train/81355.jpeg   
82210  images/train/82210.jpeg   
82918  images/train/82918.jpeg   
84674  images/train/84674.jpeg   
90157  images/train/90157.jpeg   

                                           product_title  class_label kurta  \
81355  Lucknow Chikankari Kurta Top Blouse Short Kurt...  women_kurta     0   
82210  Lucknow Chikankari Kurta Top Blouse Short Kurt...  women_kurta     0   
82918  Kota Doria Handmade Printed Cotton Saree with ...  women_kurta     0   
84674  Kota Doria Handmade Printed Cotton Saree with ...  women_kurta     0   
90157                     by FBB Sequined Festive Blouse  women_kurta     0   

      dhoti_pants  saree sherwanis palazzos nehru_jackets petticoats  blouse  \
81355           0      0         0        0             0          0  blouse   
82210           0      0         0        0             0          0  blouse   
82918           0  saree         0        0             0          0  blouse   
84674           0  saree         0        0             0          0  blouse   
90157           0      0         0        0             0          0  blouse   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
81355        0     0       0           0         0           0             0   
82210        0     0       0           0         0           0             0   
82918        0     0       0           0         0           0             0   
84674        0     0       0           0         0           0             0   
90157        0     0       0           0         0           0             0   

      leggings_and_salwars  
81355                    0  
82210                    0  
82918                    0  
84674                    0  
90157                    0

In [139]:
df_blouse.iloc[1,1], df_blouse.iloc[1,2]

('Blue Chanderi Pant Saree without Blouse (Fully-Stitched)…', 'dhoti_pants')

### Mislabeled:

In [140]:
df_blouse.iloc[3,1], df_blouse.iloc[3,2]

('Womens Raglan Sleeve Body Blouse Top', 'dhoti_pants')

### Mislabeled:

In [141]:
df_blouse.iloc[4,1], df_blouse.iloc[4,2]

('Girls Raglan Sleeve Body Blouse Top', 'dhoti_pants')

### Dhoti Pants

In [142]:
df_dhoti_pants = df_revised[df_revised.dhoti_pants == 'dhoti_pants']
df_dhoti_pants.shape

(427, 19)

In [143]:
df_dhoti_pants.head()

image_path  \
1277    images/train/1277.jpeg   
2398    images/train/2398.jpeg   
2598    images/train/2598.jpeg   
27759  images/train/27759.jpeg   
28621  images/train/28621.jpeg   

                                           product_title class_label kurta  \
1277   Women's VISCOSE RAYON Dhoti Saree (MUSTARD YEL...       saree     0   
2398   chios Women Cotton Dhoti Saree Without Blouse ...       saree     0   
2598             Women's Viscose Rayon Dhoti Saree (Red)       saree     0   
27759  Women's Olive Green Poly Crepe Crop Top And Dh...    dupattas     0   
28621  Women's Pure Cotton Plain Semi Patiala Dhoti S...    dupattas     0   

       dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
1277   dhoti_pants     0         0        0             0          0      0   
2398   dhoti_pants     0         0        0             0          0      0   
2598   dhoti_pants     0         0        0             0          0      0   
27759  dhoti_pants     0         0        0             0          0      0   
28621  dhoti_pants     0         0        0             0          0      0   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
1277         0     0       0           0         0           0             0   
2398         0     0       0           0         0           0             0   
2598         0     0       0           0         0           0             0   
27759        0     0       0           0         0           0             0   
28621        0     0       0           0         0           0             0   

       leggings_and_salwars  
1277                      0  
2398                      0  
2598                      0  
27759                     0  
28621  leggings_and_salwars

In [144]:
df_dhoti_pants.iloc[0,1]

"Women's VISCOSE RAYON Dhoti Saree (MUSTARD YELLOW)"

### Dupattas

In [145]:
df_dupattas = df_revised[df_revised.dupattas == 'dupattas']
df_dupattas.shape

(3666, 19)

In [146]:
df_dupattas.head()

image_path  \
134    images/train/134.jpeg   
1622  images/train/1622.jpeg   
1623  images/train/1623.jpeg   
2455  images/train/2455.jpeg   
3023  images/train/3023.jpeg   

                                          product_title class_label kurta  \
134                           Woman's kota silk dupatta       saree     0   
1622  Lami- 9 Meter Red Jari 3D Flower Patch Embroid...       saree     0   
1623  Lami - 9 MTS. Gajri Resham Embroidery Border &...       saree     0   
2455  Brocade Saree Falls LAce Border Reel Jacquard ...       saree     0   
3023  Lami 9 Mtr Green Multicolor Flower Design Pars...       saree     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
134            0     0         0        0             0          0      0   
1622           0     0         0        0             0          0      0   
1623           0     0         0        0             0          0      0   
2455           0     0         0        0             0          0      0   
3023           0     0         0        0             0          0      0   

      dupattas gowns  lehenga women_kurta kurta_men mojaris_men mojaris_women  \
134   dupattas     0        0           0         0           0             0   
1622  dupattas     0  lehenga           0         0           0             0   
1623  dupattas     0  lehenga           0         0           0             0   
2455  dupattas     0  lehenga           0         0           0             0   
3023  dupattas     0  lehenga           0         0           0             0   

     leggings_and_salwars  
134                     0  
1622                    0  
1623                    0  
2455                    0  
3023                    0

### Mislabeled:

In [147]:
df_dupattas.iloc[0,1], df_dupattas.iloc[0,2]

("Woman's kota silk dupatta", 'saree')

In [148]:
df_gowns = df_revised[df_revised.gowns == 'gowns']
df_gowns.shape

(1979, 19)

In [149]:
df_gowns.head()

image_path  \
1483  images/train/1483.jpeg   
2497  images/train/2497.jpeg   
2961  images/train/2961.jpeg   
3027  images/train/3027.jpeg   
3829  images/train/3829.jpeg   

                                          product_title class_label kurta  \
1483   Rosegold Georgette Embellished Saree Style Dress       saree     0   
2497  9 mtr pink color Laces Heavy Embroderi work re...       saree     0   
2961  Generic Black Heavy Laces for Dresses Sarees D...       saree     0   
3027                     Women Regular Color Rayon Gown       saree     0   
3829      Black Georgette Embellished Saree Style Dress       saree     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
1483           0     0         0        0             0          0      0   
2497           0     0         0        0             0          0      0   
2961           0     0         0        0             0          0      0   
3027           0     0         0        0             0          0      0   
3829           0     0         0        0             0          0      0   

     dupattas  gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
1483        0  gowns       0           0         0           0             0   
2497        0  gowns       0           0         0           0             0   
2961        0  gowns       0           0         0           0             0   
3027        0  gowns       0           0         0           0             0   
3829        0  gowns       0           0         0           0             0   

     leggings_and_salwars  
1483                    0  
2497                    0  
2961                    0  
3027                    0  
3829                    0

In [150]:
df_gowns.iloc[1,1], df_gowns.iloc[1,2]

('9 mtr pink color Laces Heavy Embroderi work real mirror Border for Dresses, Sarees,',
 'saree')

### Mislabeled:

In [151]:
df_gowns.iloc[3,1], df_gowns.iloc[3,2]

('Women Regular Color Rayon Gown', 'saree')

In [152]:
df_kurta = df_revised[df_revised.kurta == 'kurta']
df_kurta.shape

(138, 19)

In [153]:
df_kurta.head()

image_path  \
2014  images/train/2014.jpeg   
4799  images/train/4799.jpeg   
5246  images/train/5246.jpeg   
5480  images/train/5480.jpeg   
6885  images/train/6885.jpeg   

                                          product_title class_label  kurta  \
2014  Handloom Pure Cotton Jharna Khadi Saree Kurta ...       saree  kurta   
4799      Couple Set Cotton Silk saree with kurta combo       saree  kurta   
5246        Handloom Pure Cotton Saree Kurta Couple Set       saree  kurta   
5480  Handloom Pure Cotton Jharna Khadi Saree Kurta ...       saree  kurta   
6885        Handloom Pure Cotton Saree Kurta Couple Set       saree  kurta   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
2014           0     0         0        0             0          0      0   
4799           0     0         0        0             0          0      0   
5246           0     0         0        0             0          0      0   
5480           0     0         0        0             0          0      0   
6885           0     0         0        0             0          0      0   

     dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
2014        0     0       0           0         0           0             0   
4799        0     0       0           0         0           0             0   
5246        0     0       0           0         0           0             0   
5480        0     0       0           0         0           0             0   
6885        0     0       0           0         0           0             0   

     leggings_and_salwars  
2014                    0  
4799                    0  
5246                    0  
5480                    0  
6885                    0

### Mislabeled:

In [154]:
df_kurta.iloc[1,1], df_kurta.iloc[1,2]

('Couple Set Cotton Silk saree with kurta combo', 'saree')

In [155]:
df_kurta_men = df_revised[df_revised.kurta_men == 'kurta_men']
df_kurta_men.shape

(437, 19)

In [156]:
df_kurta_men.head()

image_path  \
21949  images/train/21949.jpeg   
21967  images/train/21967.jpeg   
22158  images/train/22158.jpeg   
22301  images/train/22301.jpeg   
22327  images/train/22327.jpeg   

                                           product_title  class_label kurta  \
21949  NEW GEN Boys Full Sleev Jacket Kurta & DHUTI P...  dhoti_pants     0   
21967  Kids Ethnic Yellow Krishna Dress Dhoti and Kur...  dhoti_pants     0   
22158  Boys' Black Cotton Blend Sherwani Style Kurta ...  dhoti_pants     0   
22301  Kids Boys Festive & Party Full Sleeves Kurta &...  dhoti_pants     0   
22327  Men's Ethnic Poly Cotton Full Sleeve Checkered...  dhoti_pants     0   

      dhoti_pants saree  sherwanis palazzos nehru_jackets petticoats blouse  \
21949           0     0          0        0             0          0      0   
21967           0     0          0        0             0          0      0   
22158           0     0  sherwanis        0             0          0      0   
22301           0     0          0        0             0          0      0   
22327           0     0          0        0             0          0      0   

      dupattas  gowns lehenga women_kurta  kurta_men mojaris_men  \
21949        0      0       0           0  kurta_men           0   
21967        0  gowns       0           0  kurta_men           0   
22158        0      0       0           0  kurta_men           0   
22301        0      0       0           0  kurta_men           0   
22327        0      0       0           0  kurta_men           0   

      mojaris_women leggings_and_salwars  
21949             0                    0  
21967             0                    0  
22158             0                    0  
22301             0                    0  
22327             0                    0

### Mislabeled:

In [157]:
df_kurta_men.iloc[0,1], df_kurta_men.iloc[0,2]

('NEW GEN Boys Full Sleev Jacket Kurta & DHUTI Pant (Orange_6-12M)',
 'dhoti_pants')

In [158]:
df_leggings_and_salwars = df_revised[df_revised.leggings_and_salwars == 'leggings_and_salwars']
df_leggings_and_salwars.shape

(2291, 19)

In [159]:
df_leggings_and_salwars.head()

image_path  \
330    images/train/330.jpeg   
2776  images/train/2776.jpeg   
4629  images/train/4629.jpeg   
4891  images/train/4891.jpeg   
6033  images/train/6033.jpeg   

                                          product_title class_label kurta  \
330   Women's Fernandez Sequence Georgette Saree (Sa...       saree     0   
2776  Women's Cotton Silk Blend Embroidered Salwar S...       saree     0   
4629  Women's Italian Silk Crepe Teachers Saree Salw...       saree     0   
4891  Women's Gold Sanna Silk With Embroidery Sequen...       saree     0   
6033        Women's Cotton princess cut Salwar Suit Set       saree     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
330            0     0         0        0             0          0      0   
2776           0     0         0        0             0          0      0   
4629           0     0         0        0             0          0      0   
4891           0     0         0        0             0          0      0   
6033           0     0         0        0             0          0      0   

     dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
330         0     0       0           0         0           0             0   
2776        0     0       0           0         0           0             0   
4629        0     0       0           0         0           0             0   
4891        0     0       0           0         0           0             0   
6033        0     0       0           0         0           0             0   

      leggings_and_salwars  
330   leggings_and_salwars  
2776  leggings_and_salwars  
4629  leggings_and_salwars  
4891  leggings_and_salwars  
6033  leggings_and_salwars

### Correct labels:

In [160]:
df_leggings_and_salwars.iloc[0,1], df_leggings_and_salwars.iloc[0,2]

("Women's Fernandez Sequence Georgette Saree (Salwar_ER123150_Yellow)",
 'saree')

In [161]:
df_leggings_and_salwars.iloc[4,1]

"Women's Cotton princess cut Salwar Suit Set"

In [162]:
df_lehenga = df_revised[df_revised.lehenga == 'lehenga']
df_lehenga.shape

(312, 19)

In [163]:
df_lehenga.head()

image_path  \
1622  images/train/1622.jpeg   
1623  images/train/1623.jpeg   
2235  images/train/2235.jpeg   
2382  images/train/2382.jpeg   
2455  images/train/2455.jpeg   

                                          product_title class_label kurta  \
1622  Lami- 9 Meter Red Jari 3D Flower Patch Embroid...       saree     0   
1623  Lami - 9 MTS. Gajri Resham Embroidery Border &...       saree     0   
2235  Women's Silk Heavy Work Readymade Blouse for S...       saree     0   
2382  Women's Net Semi-stitched Lehenga Choli (67-IM...       saree     0   
2455  Brocade Saree Falls LAce Border Reel Jacquard ...       saree     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
1622           0     0         0        0             0          0      0   
1623           0     0         0        0             0          0      0   
2235           0     0         0        0             0          0      0   
2382           0     0         0        0             0          0      0   
2455           0     0         0        0             0          0      0   

      dupattas gowns  lehenga women_kurta kurta_men mojaris_men mojaris_women  \
1622  dupattas     0  lehenga           0         0           0             0   
1623  dupattas     0  lehenga           0         0           0             0   
2235         0     0  lehenga           0         0           0             0   
2382         0     0  lehenga           0         0           0             0   
2455  dupattas     0  lehenga           0         0           0             0   

     leggings_and_salwars  
1622                    0  
1623                    0  
2235                    0  
2382                    0  
2455                    0

### Mislabeled:

In [164]:
df_lehenga.iloc[3,1], df_lehenga.iloc[3,2]

("Women's Net Semi-stitched Lehenga Choli (67-IM5V-TOTL_Multicolored_Free Size)",
 'saree')

In [165]:
df_mojaris_men = df_revised[df_revised.mojaris_men == 'mojaris_men']
df_mojaris_men.shape

(102, 19)

In [166]:
df_mojaris_men.head()

image_path  \
58976  images/train/58976.jpeg   
59077  images/train/59077.jpeg   
59141  images/train/59141.jpeg   
59145  images/train/59145.jpeg   
59320  images/train/59320.jpeg   

                                           product_title    class_label kurta  \
58976                                       Boy's Mojari  mojaris_women     0   
59077  Boys Designer Jutti/Mojari for Kids Ethnic Foo...  mojaris_women     0   
59141                                       Men's Mojari  mojaris_women     0   
59145  KADAM Men White Kolhapuri Design Leather Ethni...  mojaris_women     0   
59320                              Men's Leather Mojaris  mojaris_women     0   

      dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
58976           0     0         0        0             0          0      0   
59077           0     0         0        0             0          0      0   
59141           0     0         0        0             0          0      0   
59145           0     0         0        0             0          0      0   
59320           0     0         0        0             0          0      0   

      dupattas gowns lehenga women_kurta kurta_men  mojaris_men mojaris_women  \
58976        0     0       0           0         0  mojaris_men             0   
59077        0     0       0           0         0  mojaris_men             0   
59141        0     0       0           0         0  mojaris_men             0   
59145        0     0       0           0         0  mojaris_men             0   
59320        0     0       0           0         0  mojaris_men             0   

      leggings_and_salwars  
58976                    0  
59077                    0  
59141                    0  
59145                    0  
59320                    0

### Many of the images labeled mojaris_women are actually men's

In [167]:
df_mojaris_women = df_revised[df_revised.mojaris_women == 'mojaris_women']
df_mojaris_women.shape

(16, 19)

In [168]:
df_mojaris_women.head()

image_path  \
56696  images/train/56696.jpeg   
56738  images/train/56738.jpeg   
56958  images/train/56958.jpeg   
57048  images/train/57048.jpeg   
57457  images/train/57457.jpeg   

                                           product_title  class_label kurta  \
56696          Ladies Jutti Women Jutti Women Mojari Tan  mojaris_men     0   
56738  Ladies Jutti Women Jutti Women Mojari Light Brown  mojaris_men     0   
56958                                     Womens Mojaris  mojaris_men     0   
57048  Men's/Women's Winter Woollen House Bedroom Ind...  mojaris_men     0   
57457  Twinkle Star Juttis, Mojaris, Khussa Shoes for...  mojaris_men     0   

      dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
56696           0     0         0        0             0          0      0   
56738           0     0         0        0             0          0      0   
56958           0     0         0        0             0          0      0   
57048           0     0         0        0             0          0      0   
57457           0     0         0        0             0          0      0   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men  mojaris_women  \
56696        0     0       0           0         0           0  mojaris_women   
56738        0     0       0           0         0           0  mojaris_women   
56958        0     0       0           0         0           0  mojaris_women   
57048        0     0       0           0         0           0  mojaris_women   
57457        0     0       0           0         0           0  mojaris_women   

      leggings_and_salwars  
56696                    0  
56738                    0  
56958                    0  
57048                    0  
57457                    0

In [169]:
df_mojaris_women.iloc[4,1]

'Twinkle Star Juttis, Mojaris, Khussa Shoes for Women'

### Many of the images labeled mojaris_men are actually women's

In [170]:
df_nehru_jackets = df_revised[df_revised.nehru_jackets == 'nehru_jackets']
df_nehru_jackets.shape

(148, 19)

In [171]:
df_nehru_jackets.head()

image_path  \
20012  images/train/20012.jpeg   
23003  images/train/23003.jpeg   
36891  images/train/36891.jpeg   
36898  images/train/36898.jpeg   
37148  images/train/37148.jpeg   

                                           product_title  class_label kurta  \
20012   Men's Nehru Jacket (VASMJ011_44_Multicolored_44)  dhoti_pants     0   
23003      Men's Kurta Pyjama & Printed Nehru Jacket set  dhoti_pants     0   
36891  Printed Achkan Style Full Sleeves Kurta and Ch...    kurta_men     0   
36898  Elegant Floral Print Sytlish Kurta and Churida...    kurta_men     0   
37148  Printed Full Sleeves Kurta and Churidar Set wi...    kurta_men     0   

      dhoti_pants saree sherwanis palazzos  nehru_jackets petticoats blouse  \
20012           0     0         0        0  nehru_jackets          0      0   
23003           0     0         0        0  nehru_jackets          0      0   
36891           0     0         0        0  nehru_jackets          0      0   
36898           0     0         0        0  nehru_jackets          0      0   
37148           0     0         0        0  nehru_jackets          0      0   

      dupattas gowns lehenga women_kurta  kurta_men mojaris_men mojaris_women  \
20012        0     0       0           0          0           0             0   
23003        0     0       0           0  kurta_men           0             0   
36891        0     0       0           0          0           0             0   
36898        0     0       0           0          0           0             0   
37148        0     0       0           0          0           0             0   

      leggings_and_salwars  
20012                    0  
23003                    0  
36891                    0  
36898                    0  
37148                    0

### Mislabeled:

In [173]:
df =  df_nehru_jackets[df_nehru_jackets.class_label == 'dhoti_pants']

In [174]:
df.head()

image_path  \
20012  images/train/20012.jpeg   
23003  images/train/23003.jpeg   

                                          product_title  class_label kurta  \
20012  Men's Nehru Jacket (VASMJ011_44_Multicolored_44)  dhoti_pants     0   
23003     Men's Kurta Pyjama & Printed Nehru Jacket set  dhoti_pants     0   

      dhoti_pants saree sherwanis palazzos  nehru_jackets petticoats blouse  \
20012           0     0         0        0  nehru_jackets          0      0   
23003           0     0         0        0  nehru_jackets          0      0   

      dupattas gowns lehenga women_kurta  kurta_men mojaris_men mojaris_women  \
20012        0     0       0           0          0           0             0   
23003        0     0       0           0  kurta_men           0             0   

      leggings_and_salwars  
20012                    0  
23003                    0

In [ ]:
df_nehru_jackets.iloc[0,1], df_nehru_jackets.iloc[0,2]

("Men's Nehru Jacket (VASMJ011_44_Multicolored_44)", 'dhoti_pants')

In [ ]:
df_palazzos = df_revised[df_revised.palazzos == 'palazzos']
df_palazzos.shape

(3145, 19)

In [ ]:
df_palazzos.head()

image_path  \
5974    images/train/5974.jpeg   
12412  images/train/12412.jpeg   
19980  images/train/19980.jpeg   
20016  images/train/20016.jpeg   
20038  images/train/20038.jpeg   

                                           product_title  class_label kurta  \
5974   HUNNY- BUNCH Premium Soft Uppada Silk Fabric U...        saree     0   
12412  Omi Fashion Women's Ethnic Jacket, Blouse and ...       blouse     0   
19980                    Women's Palazzo Wide Leg Bottom  dhoti_pants     0   
20016                             Women's Palazzo Bottom  dhoti_pants     0   
20038  Women Rayon Flared Garara/Sharara Palazzo Pant...  dhoti_pants     0   

      dhoti_pants saree sherwanis  palazzos nehru_jackets petticoats blouse  \
5974            0     0         0  palazzos             0          0      0   
12412           0     0         0  palazzos             0          0      0   
19980           0     0         0  palazzos             0          0      0   
20016           0     0         0  palazzos             0          0      0   
20038           0     0         0  palazzos             0          0      0   

      dupattas  gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
5974         0  gowns       0           0         0           0             0   
12412        0      0       0           0         0           0             0   
19980        0      0       0           0         0           0             0   
20016        0      0       0           0         0           0             0   
20038        0      0       0           0         0           0             0   

      leggings_and_salwars  
5974                     0  
12412                    0  
19980                    0  
20016                    0  
20038                    0

### Both mislabeled:

In [ ]:
df_palazzos.iloc[2,1], df_palazzos.iloc[2,2]

("Women's Palazzo Wide Leg Bottom", 'dhoti_pants')

In [ ]:
df_palazzos.iloc[2,1]

"Women's Palazzo Wide Leg Bottom"

In [ ]:
df_petticoats = df_revised[df_revised.petticoats == 'petticoats']
df_petticoats.shape

(63, 19)

In [ ]:
df_petticoats.head()

image_path                                      product_title  \
227  images/train/227.jpeg  Women's Fine Knitted Sexy Inskirt Saree Pettic...   
845  images/train/845.jpeg  Women's Yellow Shapewear Lycra Stretchable Sli...   
871  images/train/871.jpeg                  Women's Saree Shapewear Petticoat   
879  images/train/879.jpeg  Women's Pure Cotton Readymade Inskirt Saree Pe...   
890  images/train/890.jpeg  Women's Wine Stretchable Slim Fit Saree Shapew...   

    class_label kurta dhoti_pants saree sherwanis palazzos nehru_jackets  \
227       saree     0           0     0         0        0             0   
845       saree     0           0     0         0        0             0   
871       saree     0           0     0         0        0             0   
879       saree     0           0     0         0        0             0   
890       saree     0           0     0         0        0             0   

     petticoats blouse dupattas gowns lehenga women_kurta kurta_men  \
227  petticoats      0        0     0       0           0         0   
845  petticoats      0        0     0       0           0         0   
871  petticoats      0        0     0       0           0         0   
879  petticoats      0        0     0       0           0         0   
890  petticoats      0        0     0       0           0         0   

    mojaris_men mojaris_women leggings_and_salwars  
227           0             0                    0  
845           0             0                    0  
871           0             0                    0  
879           0             0                    0  
890           0             0                    0

In [ ]:
df_petticoats.iloc[0,1], df_petticoats.iloc[0,2]

("Women's Fine Knitted Sexy Inskirt Saree Petticoats with Lace (Free Size)",
 'saree')

### Both are mislabeled:

In [ ]:
df_petticoats.iloc[2,1], df_petticoats.iloc[2,1], 

("Women's Saree Shapewear Petticoat", "Women's Saree Shapewear Petticoat")

In [ ]:
df_sherwanis = df_revised[df_revised.sherwanis == 'sherwanis']
df_sherwanis.shape

(286, 19)

In [ ]:
df_sherwanis.head()

image_path  \
22158  images/train/22158.jpeg   
23016  images/train/23016.jpeg   
23066  images/train/23066.jpeg   
23210  images/train/23210.jpeg   
23211  images/train/23211.jpeg   

                                           product_title  class_label kurta  \
22158  Boys' Black Cotton Blend Sherwani Style Kurta ...  dhoti_pants     0   
23016  Men's Velvet Harem Dhoti Pants for Sherwani & ...  dhoti_pants     0   
23066  Men's Velvet Harem Dhoti Pants for Sherwani & ...  dhoti_pants     0   
23210  Men's Velvet Ready to Wear Free Size Harem Dho...  dhoti_pants     0   
23211  Men Brown Cotton Silk Sherwani and Cow Style D...  dhoti_pants     0   

      dhoti_pants saree  sherwanis palazzos nehru_jackets petticoats blouse  \
22158           0     0  sherwanis        0             0          0      0   
23016           0     0  sherwanis        0             0          0      0   
23066           0     0  sherwanis        0             0          0      0   
23210           0     0  sherwanis        0             0          0      0   
23211           0     0  sherwanis        0             0          0      0   

      dupattas gowns lehenga women_kurta  kurta_men mojaris_men mojaris_women  \
22158        0     0       0           0  kurta_men           0             0   
23016        0     0       0           0          0           0             0   
23066        0     0       0           0          0           0             0   
23210        0     0       0           0  kurta_men           0             0   
23211        0     0       0           0          0           0             0   

      leggings_and_salwars  
22158                    0  
23016                    0  
23066                    0  
23210                    0  
23211                    0

### Also mislabeled. It is a sherwani-style kurta, not a sherwani

In [ ]:
df_sherwanis.iloc[0,1], df_sherwanis.iloc[0,2]

("Boys' Black Cotton Blend Sherwani Style Kurta Set(VASBSW120)", 'dhoti_pants')

In [ ]:
df_women_kurta = df_revised[df_revised.women_kurta == 'women_kurta']
df_women_kurta.shape

(1711, 19)

In [ ]:
df_women_kurta.head()

image_path  \
1112  images/train/1112.jpeg   
1865  images/train/1865.jpeg   
2821  images/train/2821.jpeg   
3683  images/train/3683.jpeg   
4228  images/train/4228.jpeg   

                                          product_title class_label kurta  \
1112  Women's Khadi Handloom Cotton Mens & Womens Co...       saree     0   
1865                       Women's Crepe Straight Kurta       saree     0   
2821  Women's Khadi Handloom Cotton Mens & Womens Co...       saree     0   
3683  Women's Summer Cotton Mens & Womens Couple Set...       saree     0   
4228  Women's Khadi Handloom Cotton Mens & Womens Co...       saree     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
1112           0     0         0        0             0          0      0   
1865           0     0         0        0             0          0      0   
2821           0     0         0        0             0          0      0   
3683           0     0         0        0             0          0      0   
4228           0     0         0        0             0          0      0   

     dupattas gowns lehenga  women_kurta kurta_men mojaris_men mojaris_women  \
1112        0     0       0  women_kurta         0           0             0   
1865        0     0       0  women_kurta         0           0             0   
2821        0     0       0  women_kurta         0           0             0   
3683        0     0       0  women_kurta         0           0             0   
4228        0     0       0  women_kurta         0           0             0   

     leggings_and_salwars  
1112                    0  
1865                    0  
2821                    0  
3683                    0  
4228                    0

### Mislabeled:

In [ ]:
df_women_kurta.iloc[1,1], df_women_kurta.iloc[1,2]

("Women's Crepe Straight Kurta", 'saree')

In [ ]:
df_revised.to_csv('/Users/butler/Documents/Image-Classification-Indian-Clothing/data/processed/train_other_categories.csv')

## While not all of the items in each dataframe are mislabeled, many are. 
## There are 1723 rows in the saree dataframe;
## 2337 rows in the blouse dataframe;
## 427 rows in the dhoti_pants dataframe;
## 3666 rows in the dupatta dataframe;
## 1979 rows in the gown dataframe;
## 138 rows in the kurta dataframe;
## 416 rows in the kurta_men dataframe;
## 2291 rows in the leggings_and_salwars dataframe;
## 2337 rows in the blouse dataframe;
## 312 rows in the lehenga dataframe;
## 96 rows in the mojaris_men dataframe;
## 13 rows in the mojaris_women dataframe;
## 148 rows in the Nehru_jackets dataframe;
## 3145 rows in the palazzos dataframe;
## 63 rows in the petticoats dataframe;
## 286 rows in the sherwani dataframe;
## and 1654 rows in the women_kurta dataframe.